generators/generatorPickerFunctions.cc

```diff
@@ -176,12 +185,25 @@ bool uniformMassExponentialTPicker::operator() (double& invariantMass, double& t
 		printErr << "trying to use an uninitialized massAndTPrimePicker." << endl;
 		return false;
 	}
+	TRandom3* randomNumbers = randomNumberGenerator::instance()->getGenerator();
+	invariantMass = randomNumbers->Uniform(_massRange.first, _massRange.second);
+	if (not pickTPrimeForMass(invariantMass, tPrime)) {
+		printErr << "error while generating t'." << std::endl;
+		return false;
+	}
+	return true;
+}
+
+
+bool uniformMassExponentialTPicker::pickTPrimeForMass(const double invariantMass, double& tPrime) {
```

Comment:
maybe one could improve the naming here:
`uniformMassMultiExponentialTPrimePicker` would probably be more appropriate.
Not sure though, whether this would be worth breaking the user side.